### PR TITLE
fuse ssm state store into chunk_gated_delta_rule_fwd_h

### PR DESCRIPTION
**Pull request by yizhang2077**
## Motivation

fuse following ops into `chunk_gated_delta_rule_fwd_h` kernel and do update in-place
1. `index_select` from ssm states
2.  alloc `final_state` in `chunk_gated_delta_rule_fwd_h`
3.  clone `last_recurrent_state` to transfer to ssm states type
4.  copy `last_recurrent_state` to ssm states
## Accuracy Tests

both of tests can be aligned with main

``` python
python3 -m sglang.launch_server --model Qwen/Qwen3-Next-80B-A3B-Instruct --tp 2 --disable-radix-cache --attention-backend triton

python3 benchmark/gsm8k/bench_sglang.py --num-questions 1000
Accuracy: 0.944
Invalid: 0.000
Latency: 119.545 s
Output throughput: 1394.582 token/s

python3 -m sglang.test.send_one  --return-logprob --max-new-tokens 10
return_logits: [[-0.7937768697738647, 22555, None], [-0.00028236693469807506, 0, None], [-0.4740985035896301, 21193, None], [-8.106198947643861e-06, 374, None], [-7.152555099310121e-07, 264, None], [-0.16030088067054749, 7225, None], [-6.806619057897478e-05, 15629, None], [-0.40155768394470215, 17288, None], [-8.106198947643861e-06, 7082, None], [-1.1205610462639015e-05, 3538, None]]
```
## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).
- [ ] Work with maintainers to merge your PR. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process)

### CODE REVIEW

File: python/sglang/srt/layers/attention/fla/chunk_delta_h.py [hunk 1]
```diff
@@ -90,10 +90,14 @@ def chunk_gated_delta_rule_fwd_kernel_h_blockdim64(
     stride_h = H * K * V
     stride_k = Hg * K
     stride_w = H * K
+
+    indice = tl.load(initial_state_indices + i_n).to(tl.int32)
```
Comment (ispobock) [inline python/sglang/srt/layers/attention/fla/chunk_delta_h.py]: tiny: `indice` -> `index` or `state_index`

File: python/sglang/srt/layers/attention/fla/chunk_delta_h.py [hunk 2]
```diff
@@ -90,10 +90,14 @@ def chunk_gated_delta_rule_fwd_kernel_h_blockdim64(
...
+    h0 = initial_state + indice * stride_h
+    ht = initial_state + indice * stride_h
```
Comment (ispobock) [inline python/sglang/srt/layers/attention/fla/chunk_delta_h.py]: ht = h0?

File: python/sglang/srt/layers/attention/fla/chunk_delta_h.py [hunk 3]
```diff
@@ -274,6 +278,7 @@ def chunk_gated_delta_rule_fwd_h(
     g: Optional[torch.Tensor] = None,
     gk: Optional[torch.Tensor] = None,
     initial_state: Optional[torch.Tensor] = None,
+    initial_state_indices: Optional[torch.Tensor] = None,
     output_final_state: bool = False,
```
Comment (ispobock) [inline python/sglang/srt/layers/attention/fla/chunk_delta_h.py]: `output_final_state` seems not needed?

File: python/sglang/srt/layers/attention/fla/chunk_delta_h.py [hunk 4]
```diff
@@ -330,10 +332,10 @@ def grid(meta):
         USE_G=g is not None,
         USE_GK=gk is not None,
         USE_INITIAL_STATE=initial_state is not None,
-        STORE_FINAL_STATE=final_state is not None,
+        STORE_FINAL_STATE=True,
```
Comment (ispobock) [inline python/sglang/srt/layers/attention/fla/chunk_delta_h.py]: do we need to keep `store_final_state` option?

Comment (ispobock) [reply python/sglang/srt/layers/attention/fla/chunk_delta_h.py]: Maybe we need to pass a param like `inplace_update` to let users know the input is mutable.

File: python/sglang/srt/layers/attention/fla/chunk_delta_h.py (hunk 2 above)
Comment (yizhang2077) [reply python/sglang/srt/layers/attention/fla/chunk_delta_h.py]: yes, since we use inplace update, ht position is actually h0, I keep ht here to make less difference

File: python/sglang/srt/layers/attention/fla/chunk_delta_h.py (hunk 3 above)
Comment (yizhang2077) [reply python/sglang/srt/layers/attention/fla/chunk_delta_h.py]: removed

File: python/sglang/srt/layers/attention/fla/chunk_delta_h.py (hunk 4 above)
Comment (yizhang2077) [reply python/sglang/srt/layers/attention/fla/chunk_delta_h.py]: done

**Review by ispobock — APPROVED** [review]

**Comment by hnyls2002** [comment]
https://github.com/sgl-project/sglang/actions/runs/20500409112
